src/executor/executor.cc

```diff
@@ -239,6 +229,22 @@ struct Executor::Impl {
     context.smSemaphores = std::move(smSemaphores);
     context.proxySemaphores = std::move(proxySemaphores);
 
+    auto getTransportFlags = [&](std::vector<ChannelInfo>& infos, int rank) {
```

Comment:
Seems we have dup code for `getTransportFlags` and `getBufferSize`? We already have related code at `setupRegisteredMemories`. Maybe we make functions for this, instead of lambda expressions